x-pack/platform/plugins/shared/fleet/server/services/agent_policies/package_policies_to_agent_inputs.ts

```diff
@@ -133,11 +137,15 @@ export const getFullInputStreams = (
               const streamId = stream.id;
               const { data_stream: compiledDataStream, ...compiledStream } =
                 stream.compiled_stream ?? {};
+              const dsTypeVar = stream.vars?.[DATA_STREAM_TYPE_VAR_NAME]?.value;
+              const datasetVar = stream.vars?.[DATASET_VAR_NAME]?.value;
               const fullStream: FullAgentPolicyInputStream = {
                 id: streamId,
                 data_stream: {
                   ...stream.data_stream,
                   ...compiledDataStream,
+                  ...(dsTypeVar ? { type: dsTypeVar } : {}),
+                  ...(datasetVar ? { dataset: datasetVar } : {}),
```

Comment:
I have a question about these two lines: they are not added conditionally, so this change will be applied to any type of package. 

Is there a case when this could be a problem, i.e. a regular package (not `input` type, not `otelcol`) that might have issues because of this change in the configuration?

Reply:
This can be an issue with any package that doesn't include the `data_stream` section in their agent configuration. This is the case of OTel input packages but could be the case of other packages too.

Though maybe you are right and can be risky to modify this for all packages. Other packages will have other issues if they don't include the `data_stream` section.

I will change this.

Reply:
Done in https://github.com/elastic/kibana/pull/236016/commits/23e0245b9dd7b7f051dade0d24902664d5cb8521.